lib/spack/llnl/util/filesystem.py

```diff
@@ -466,29 +471,109 @@ def fix_darwin_install_name(path):
                         stdout=subprocess.PIPE).communicate()[0]
                     break
 
-# Utilities for libraries
 
+def find(root, files, recurse=True):
+    """Search for ``files`` starting from the ``root`` directory.
 
-class LibraryList(collections.Sequence):
-    """Sequence of absolute paths to libraries
+    Like GNU/BSD find but written entirely in Python.
 
-    Provides a few convenience methods to manipulate library paths and get
-    commonly used compiler flags or names
+    Examples:
+
+    .. code-block:: console
+
+       $ find /usr -name python
+
+    is equivalent to:
+
+    .. code-block:: python
+
+       >>> find('/usr', 'python')
+
+    .. code-block:: console
+
+       $ find /usr/local/bin -maxdepth 1 -name python
+
+    is equivalent to:
+
+    .. code-block:: python
+
+       >>> find('/usr/local/bin', 'python', recurse=False)
```

Comment:
I'm honestly considering just converting this to a wrapper around the system `find`. Thoughts? Either way, I think this would be a very useful command to have.